src/terminfo.cr

```diff
@@ -600,26 +600,25 @@ module Terminfo
   # :ditto:
   def initialize(*, term : String, extended = true)
     filename = nil
+    # scan all possible locations
     ::Terminfo.directories.each do |dir|
-      f1 = File.join dir, term
-      f2 = File.join dir, term[0..0], term[0..1], term
-      filename = nil
-      if File.readable? f1
-        filename = f1
-        break
-      elsif File.readable? f2
-        filename = f2
-        break
-      end
-      if filename
-        initialize path: filename, extended: extended
+      locations = [
+        File.join(dir, term),                                 # /path/to/terminfo/screen
+        File.join(dir, term[0..0], term[0..1], term),         # /path/to/terminfo/s/sc/screen
+        File.join(dir, sprintf("%x", term[0..0].bytes), term) # /path/to/terminfo/73/screen, see https://invisible-island.net/ncurses/NEWS.html#t20071117
+      ]
+      filename = locations.select{|loc| File.readable? loc}.pop?
+      break if filename
```

Comment:
Let's change the above 2 lines to:

```cr
if filename = locations.find { |loc| File.readable? loc}
  break
end
```

Also, please see whether e.g. ncurses skips unreadable files or reports an error.
If it skips them, your test for #readable? is good. If it reports an error, then maybe we should use `File.exists?` for the same behavior.

Reply:
@docelic `infocmp` appears to skip unreadable files, based on a test in a local directory with permissions forced to root-only readability:

```
❯ infocmp -A . xterm
infocmp: couldn't open terminfo file ./78/xterm.
```

Also, if my Ruby experience will serve me any good, are you OK with me using the following single line instead?

``` crystal
break if filename = locations.find { |loc| File.readable? loc}
```

Reply:
Sure. I've also updated README to reflect these additions. Will merge and release 0.7.4 as soon as you send the updated patch. Thanks!

Reply:
@docelic Should be updated now, let me know if there's any other changes!